lib/request_overrider.js

```diff
@@ -34,8 +34,12 @@ function setHeader(request, name, value) {
   request._headerNames = request._headerNames || {};
   request._removedHeader = request._removedHeader || {};
 
-  request._headers[key] = value;
-  request._headerNames[key] = name;
+  if (request.setHeader) {
```

Comment:
even node v4 has a `setHeader` method on req, but v7 could still set directly.
Could just reuse the is-node-v8 condition computed above.